### DISCUSSION TITLE
Can I speed up JBrowse2 load time with VCF and BAM files

### DISCUSSION THREAD

**Discussion by takanoriwashio** [GMOD/jbrowse-components#2525] [Q&A]
We have installed JBrowse2 in our server, and we uploaded VCF and BAM files.
But the load time with VCF and BAM files is so slow that our user can't use JBrowse2 comfortably.

We find the following FAQ.
Can this adpot not only to JBrowse1 but also to JBrowse2?

--------------------------------------------------------
Can I speed up JBrowse load time with VCF and BAM files
https://jbrowse.org/docs/faq.html#can-i-speed-up-jbrowse-load-time-with-vcf-and-bam-files

--------------------------------------------------------

**Comment by cmdcolin** [comment]
Currently that jbrowse 1 trick does not apply to jbrowse 2

The main benefit of that jbrowse 1 trick is for if you have very large index files (some bams have 100+MB bai files for example)

If the index files are a problem for you, then you can consider a couple other options in jbrowse 2

1) Change to using .csi indexes (can be done for both BAM and VCF, csi is compressed while BAM is not)
2) Change to using CRAM, CRAM is faster than BAM in general and has smaller indexes


If the issues with slowness are related to other things, we are trying to make some optimizations and fixes currently that will hopefully come soon

1) We are making a "stats estimation" PR that makes it try not to load data if too zoomed out
2) We are making some more general optimizations that will hopefully help just general usages



If you wanted to limit your tracks so that they are only visible at very zoomed in levels via the config file, for your BAM tracks you can add this to your config

```
    "displays": [
        {
          "type": "LinearAlignmentsDisplay",
          "displayId": "id-for-LinearAlignmentsDisplay",
          "pileupDisplay": {
            "type": "LinearPileupDisplay",
            "displayId": "placeholderId",
            "maxDisplayedBpPerPx": 0.2
          },
          "snpCoverageDisplay": {
            "type": "LinearSNPCoverageDisplay",
            "displayId": "placeholderId",
            "maxDisplayedBpPerPx": 0.2
          }
        }
    ]
```

would require that they are zoomed in to see the track (zoomed in enough so that each base pair is ~0.2 pixels, or 5 px per bp)

**Reply by cmdcolin** [reply]
also, if you have any characteristics on your data might be able to advise further (e.g. is it multi-sample vcf, or depth of coverage, long or short read) and version of jbrowse and whether embedded.

**Comment by takanoriwashio** [comment]
Thank you for your prompt reply, and sorry for my slow responce.

We want to show the past sequenced data which used commercial capture panels (ex. SureSelect, Twist, etc.) on Jbrowse 2.
Then we want to know which panel support which exon regions, we want to see not only BED file but also BAM file to know in details.

When we are trying to check these result in details, we zoom in, out and move repeatedly.

But when we add BAM track on Jbrose 2, it took long time to show mapped reads even if it is only one BAM file.

We changed to using CRAM, but the performace could not be improved.

Jbrose 2 is good and convinient tool to show sequenced data for the person who is not familiar with using IGV.
We would like you to improve the performace to show BAM track.


NGS: NextSeq / Miseq- Short reads.
Capture kit: SureSelect, Twist or Whole genome seq
Version of JBrowse 2: Ver. 1.4.4

Bests,

**Reply by cmdcolin** [reply]
There are some performance improvements in the latest version v1.5.2+ (which actually fixes a performance regression that existed between about v1.3.3-v1.5.1)

It is especially faster for deep short read data. Or here, I saw it go from 70 seconds to 6 on a dataset https://github.com/GMOD/jbrowse-components/pull/2523

**Reply by cmdcolin** [reply]
There is still room for improvement, but let us know how that works if you get a chance
